client/cli/src/lib.rs

```diff
@@ -174,14 +84,22 @@ where
 		.name(version.executable_name)
 		.author(version.author)
 		.about(version.description)
-		.version(full_version.as_str());
+		.version(full_version.as_str())
+		.settings(&[
+			AppSettings::GlobalVersion,
+			AppSettings::ArgsNegateSubcommands,
+			AppSettings::SubcommandsNegateReqs,
+		]);
```

Comment:
@bkchr I'm putting back this here.

I didn't want to have them in sc_cli because I thought it would be best if they are a choice of the dev who makes the app that uses sc_cli. But I noticed in polkadot and cumulus that this was error prone especially since now that RunCmd is not a variant of the subcommands anymore.

So I added more doc on the subject on this function.

This is still flexible because the dev can choose to not use this helper and use structopt's from_iter directly if they want.